src/Symfony/Component/PropertyInfo/Type.php

```diff
@@ -59,10 +59,16 @@ class Type
     /**
      * @throws \InvalidArgumentException
      */
-    public function __construct(string $builtinType, bool $nullable = false, string $class = null, bool $collection = false, self $collectionKeyType = null, self $collectionValueType = null)
+    public function __construct(/* string|array */ $builtinType, bool $nullable = false, string $class = null, bool $collection = false, self $collectionKeyType = null, self $collectionValueType = null)
     {
-        if (!\in_array($builtinType, self::$builtinTypes)) {
-            throw new \InvalidArgumentException(sprintf('"%s" is not a valid PHP type.', $builtinType));
+        if (!\is_string($builtinType) && !\is_array($builtinType)) {
+            throw new \TypeError('"$builtinType" should be a string or an array.');
```

Comment:
the message should be as close as possible to the one php8 would throw with a `string|array` union type